### DISCUSSION TITLE
How to model a radial CylinderSegment?

### DISCUSSION THREAD

**Discussion by jefflongo** [magpylib/magpylib#736] [Q&A]
How would you model a radial multipole ring? Is there a way to have the magnetization "curve" or does it need to be approximated with many small linear sections? Here is an example magnet I'd like to model:

![image](https://github.com/magpylib/magpylib/assets/8563456/4b605129-bce5-470e-9f01-8c03aa3c9b27)

**Comment by OrtnerMichael** [comment]
hi @jefflongo 

Radially magnetized segments are a bit tricky mathematically - as you suggest, I would use multiple linear sections.

However ... there is also a bit of a problem with the magnet that you show. While manufacturers will show you these pictures, in reality, such magnets do not have "homogeneous" radial magnetizations within the 4 segments. They are magnetized like this:

![image](https://github.com/magpylib/magpylib/assets/47384012/70123a80-925d-4a33-83e8-aa4a0fb36382)

which results in a highly inhomogeneous magnetization distribution.

It's unfortunately a difficult problem for modelling - because the magnetization is unknown. If you know the magnetization tool, you could split the ring into many small segements and approximate the distribution based on the imprinted field during the magnetization process.

**Reply by OrtnerMichael** [reply]
also it is my understanding is that manufacturers try to choose the magnetization in such a way that a well-placed sensor sees a mostly harmonic field as the magnet rotates.
